### ISSUE TITLE
bug version 1.40

### ISSUE THREAD

**Issue by Elihai** [terraform-providers/terraform-provider-aws#6157]
issuse:
aws_rds_cluster.default: Failed to modify RDS Cluster (demo): InvalidParameterCombination: The specified new engine version is same as current version: 9.6.9
	status code: 400, request id: 04f08c7a-3aee-4754-96f1-b1e33bcacbe6

it happened when I trying to restore the last snapshot and the snapshot already configured to with rds engine version 9.6.9 it also happens when I trying to upgrade to 10.4.

so, for now, I back to version 1.39

**Comment by asaghri** [comment]
Same problem here ! Had to switch back to 1.39 to make it work.

**Comment by namachieli** [comment]
Ran into this myself tonight.

TF v0.11.10
AWS v1.41
Aurora RDS PostrgreSQL v10.4

Error
```
* aws_rds_cluster.postgresql: 1 error(s) occurred:

* aws_rds_cluster.postgresql: Failed to modify RDS Cluster (test-netbox-postgresql-cluster): InvalidParameterCombination: The specified new engine version is same as current version: 10.4
	status code: 400, request id: 565a<...>8b9fe5
```

**Comment by DI-DaveGoodine** [comment]
changing my provider to:
```
provider "aws" {
  region = "us-east-1"
  version = "~> 1.39.0"
}
```
fixed this, as it did for others.  I was also trying to create an rds aurora postgresql 10.4 cluster from snapshot with latest aws provider and latest tf.

**Comment by bflad** [comment]
I was able to reproduce this with some new acceptance testing -- working on a fix now. 👍 

```
--- FAIL: TestAccAWSRDSCluster_SnapshotIdentifier_EngineVersion_Different (307.45s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
        	* aws_rds_cluster.test: 1 error occurred:
        	* aws_rds_cluster.test: Failed to modify RDS Cluster (tf-acc-test-5310186777500539203): InvalidParameterCombination: The specified new engine version is same as current version: 9.6.6
        	status code: 400, request id: cd9bb70c-0af8-4783-95b9-ef9fa72251d0




--- FAIL: TestAccAWSRDSCluster_SnapshotIdentifier_EngineVersion_Equal (307.49s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
        	* aws_rds_cluster.test: 1 error occurred:
        	* aws_rds_cluster.test: Failed to modify RDS Cluster (tf-acc-test-4108683443326527325): InvalidParameterCombination: The specified new engine version is same as current version: 9.6.3
        	status code: 400, request id: ee7ee554-035b-43e7-9a66-9fd57254a04e
```

**Comment by bflad** [comment]
Pull request submitted with acceptance testing to cover this going forward: #6391

**Comment by bflad** [comment]
Sorry for the trouble, folks! The fix for this has been merged and will release with version 1.43.1 of the AWS provider, likely later today. 👍

**Comment by DI-DaveGoodine** [comment]
Thank you sir!

**Comment by bflad** [comment]
This has been released in [version 1.43.1 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1431-november-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
